### PR TITLE
Feat/reward distribution

### PR DESCRIPTION
**Pull request by nrebhun**
## Summary:

- handleData.js
    - Add experience and level tracking for characters
    - Add experience distribution function for Dungeon Master
    - Refactor for ES6

### CODE REVIEW

**Comment by nrebhun** [comment]
@zmbush
When you have a spare moment!
Thank you!

After merge I plan to branch from master to create a unit test branch 👍

File: handleData.js [hunk 1]
```diff
@@ -186,6 +185,8 @@ class Character extends Player {
             "Platinum"   : 0,
             "Electrum"   : 0,
         };
+        this.currentExperience = initialExperience;
+        this.experienceRequirements = [300, 900, 2700, 5400];
```
Comment (zmbush) [inline handleData.js]: These should be constants somewhere else, these numbers aren't specific to a given character

File: handleData.js [hunk 2]
```diff
@@ -201,15 +202,7 @@ class Character extends Player {
     removeItem(targetItem) {
         if (targetItem.name in this.inventory) {
             this.inventory[targetItem.name].quantity -= targetItem.quantity;
-
-            if (this.inventory[targetItem.name].quantity <= 0) {
-                delete this.inventory[targetItem.name];
-                return InventoryResult.Removed;
-            } else {
-                return InventoryResult.UpdatedExisting;
-            }
```
Comment (zmbush) [inline handleData.js]: Merge conflict? This code seemed useful.

**Comment by zmbush** [comment]
This mostly looks reasonable :)
Just a couple nits

File: handleData.js (hunk 1 above)
Comment (nrebhun) [inline handleData.js]: How do you mean "they aren't specific to a given character?"

File: handleData.js (hunk 2 above)
Comment (nrebhun) [inline handleData.js]: Yeah, looks like I removed this accidentally. Thanks

File: handleData.js [hunk 3]
```diff
@@ -41,6 +41,13 @@ Class = {
     Wizard      : "Wizard",
 };
 
+// Constants
+const generalExpRequirements = [0, 300, 900, 2700,
+                                6500, 14000, 23000, 34000, 
+                                48000, 64000, 85000, 100000, 
+                                120000, 140000, 165000, 195000,
+                                225000, 265000, 305000, 355000,];
+
```
Comment (nrebhun) [inline handleData.js]: Something like this?

**Pull request merged**